workspaces/local-cli/src/shared/coverage-2.ts

```diff
@@ -0,0 +1,231 @@
+const TotalForPathAndMethod = createDataClass(function TotalForPathAndMethod(
+  pathId: String,
+  httpMethod: String
+) {});
+const TotalForPathAndMethodAndStatusCode = createDataClass(
+  function TotalForPathAndMethodAndStatusCode(
+    pathId: String,
+    httpMethod: String,
+    httpStatusCode: number
+  ) {}
+);
+const TotalForPathAndMethodWithoutBody = createDataClass(
+  function TotalForPathAndMethodWithoutBody(
+    pathId: String,
+    httpMethod: String
+  ) {}
+);
+const TotalForPathAndMethodAndContentType = createDataClass(
+  function TotalForPathAndMethodAndContentType(
+    pathId: String,
+    httpMethod: String,
+    requestContentType: String
+  ) {}
+);
+const TotalForPathAndMethodAndStatusCodeWithoutBody = createDataClass(
+  function TotalForPathAndMethodAndStatusCodeWithoutBody(
+    pathId: String,
+    httpMethod: String,
+    httpStatusCode: number
+  ) {}
+);
+const TotalForPathAndMethodAndStatusCodeAndContentType = createDataClass(
+  function TotalForPathAndMethodAndStatusCodeAndContentType(
+    pathId: String,
+    httpMethod: String,
+    httpStatusCode: number,
+    responseContentType: String
+  ) {}
+);
+const TotalForRequest = createDataClass(function TotalForRequest(
+  requestId: string
+) {});
+const TotalForResponse = createDataClass(function TotalForResponse(
+  responseId: string
+) {});
+
+type ICoverageMap = { [key: string]: number };
+
+const recordEvent = (evt: string, coverage_map: ICoverageMap) => {
+  coverage_map[evt] = (coverage_map[evt] ?? 0) + 1;
+};
+
+async function computeCoverage(paths: IPathMapping, captureId: string) {
+  const events = await getSpecEventsFrom(paths.specStorePath);
+  // Will this be laggy? Specs might get big :thinking_face:
+  const spec = DiffEngine.spec_from_events(JSON.stringify(events));
+
+  const captureIterator = CaptureInteractionIterator(
+    {
+      captureId,
+      captureBaseDirectory: paths.capturesPath,
+    },
+    (_) => true
+  );
+
+  const coverages = (async function* (interactionItems) {
+    for await (let item of interactionItems) {
+      if (!item.hasMoreInteractions) break;
+      if (!item.interaction) continue;
+
+      const diff_str = DiffEngine.diff_interaction(
+        JSON.stringify(item.interaction),
+        spec
+      );
+      // Maybe we could be more clever here later
+      // Would also love to get real types in here
+      const diff: Array<any> = JSON.parse(diff_str);
+
+      let is_diff = diff.length > 0;
+
+      yield {
+        coverage: interactionToCoverage(spec, item.interaction.value),
+        is_diff,
+      };
+    }
+  })(captureIterator);
+
+  let combined_diff_map: ICoverageMap = {};
+  let combined_non_diff_map: ICoverageMap = {};
+
+  for await (let { coverage: individual_coverage, is_diff } of coverages) {
+    for (const [k, v] of Object.entries(individual_coverage)) {
+      let chosen_map = is_diff ? combined_diff_map : combined_non_diff_map;
```

Comment:
Looks like this can move into the outer loop

Reply:
Done!